### ISSUE TITLE
Crash when running Realm at Watch Extension using XCode 10

### ISSUE THREAD

**Issue by ggirotto** [realm/realm-cocoa#5929]
**Bugs**: Watch app crashes when running in Apple Watch series 4 and XCode 10.

## Goals && Expected Results
Apple Watch app launches fine, just like iOS using XCode 10 and latest 3.10 Realm release.

## Actual Results
App instantly crashes when loading.
Terminal Output:

```
2018-09-23 14:20:03.276077-0300 Hercules-watchOS Extension[92585:2956772] Task <EE3009AA-79DC-49DE-A58F-7ABFCEE84760>.<1> load failed with error Error Domain=NSURLErrorDomain Code=-2000 "can’t load from network" UserInfo={NSErrorFailingURLStringKey=https://static.realm.io/update/cocoa?3.10.0, NSErrorFailingURLKey=https://static.realm.io/update/cocoa?3.10.0, NSLocalizedDescription=can’t load from network, _NSURLErrorRelatedURLSessionTaskErrorKey=(
    "LocalDataTask <EE3009AA-79DC-49DE-A58F-7ABFCEE84760>.<1>"
), NSUnderlyingError=0x7a028400 {Error Domain=kCFErrorDomainCFNetwork Code=-2000 "(null)"}, _NSURLErrorFailingURLSessionTaskErrorKey=LocalDataTask <EE3009AA-79DC-49DE-A58F-7ABFCEE84760>.<1>} [-2000]

2018-09-23 14:20:03.294693-0300 Hercules-watchOS Extension[92585:2956772] Task <4E78EE87-9751-4A16-A8E2-876795969756>.<2> load failed with error Error Domain=NSURLErrorDomain Code=-2000 "can’t load from network" UserInfo={NSErrorFailingURLStringKey=https://api.mixpanel.com/track/?data=eyJldmVudCI6IlJ1biIsInByb3BlcnRpZXMiOnsiVGFyZ2V0IE9TIE1pbmltdW0gVmVyc2lvbiI6IjQuMyIsIkJpbmRpbmciOiJjb2NvYSIsImRpc3RpbmN0X2lkIjoiZmJkMDNlZTJkNWQzZDUzODU4YzYzMjJmMzM5Y2JiOTk2Y2VkMGRmMDJlYWQ5NmE0YWM4MDZiOTM1MjBmNTMyOSIsIkhvc3QgT1MgVHlwZSI6Im9zeCIsIkFub255bWl6ZWQgTUFDIEFkZHJlc3MiOiJmYmQwM2VlMmQ1ZDNkNTM4NThjNjMyMmYzMzljYmI5OTZjZWQwZGYwMmVhZDk2YTRhYzgwNmI5MzUyMGY1MzI5IiwiVGFyZ2V0IE9TIFZlcnNpb24iOiJWZXJzaW9uIDUuMCAoQnVpbGQgMTZSMzYzKSIsIlN5bmMgVmVyc2lvbiI6IjMuOS4xMCIsIkFub255bWl6ZWQgQnVuZGxlIElEIjoiYjRiMDIyOGVkMzEwM2YzOTJmZmIxMDEyZjQ0OTY5YWFmYzJkYzAxNTE3NjgwYTkyMjQxNzA3ODQ0MTU3YWFhZSIsIkhvc3QgT1MgVmVyc2lvbiI6IjE3LjcuMCIsInRva2VuIjoiY2UwZmFjMTk1MDhmNmM4ZjIwMDY2ZDM0NWQzNjBmZDAiLCJUYXJnZXQgT1MgVHlwZSI6IndhdGNob3MiLCJMYW5ndWFnZSI6InN3aWZ0IiwiU3dpZnQgVmVyc2lvbiI6IjQuMiIsIlJlYWxtIFZlcnNpb24iOiIzLjEwLjAifX0=&ip=1, NSErrorFailingURLKey=https://api.mixpanel.com/track/?data=eyJldmVudCI6IlJ1biIsInByb3BlcnRpZXMiOnsiVGFyZ2V0IE9TIE1pbmltdW0gVmVyc2lvbiI6IjQuMyIsIkJpbmRpbmciOiJjb2NvYSIsImRpc3RpbmN0X2lkIjoiZmJkMDNlZTJkNWQzZDUzODU4YzYzMjJmMzM5Y2JiOTk2Y2VkMGRmMDJlYWQ5NmE0YWM4MDZiOTM1MjBmNTMyOSIsIkhvc3QgT1MgVHlwZSI6Im9zeCIsIkFub255bWl6ZWQgTUFDIEFkZHJlc3MiOiJmYmQwM2VlMmQ1ZDNkNTM4NThjNjMyMmYzMzljYmI5OTZjZWQwZGYwMmVhZDk2YTRhYzgwNmI5MzUyMGY1MzI5IiwiVGFyZ2V0IE9TIFZlcnNpb24iOiJWZXJzaW9uIDUuMCAoQnVpbGQgMTZSMzYzKSIsIlN5bmMgVmVyc2lvbiI6IjMuOS4xMCIsIkFub255bWl6ZWQgQnVuZGxlIElEIjoiYjRiMDIyOGVkMzEwM2YzOTJmZmIxMDEyZjQ0OTY5YWFmYzJkYzAxNTE3NjgwYTkyMjQxNzA3ODQ0MTU3YWFhZSIsIkhvc3QgT1MgVmVyc2lvbiI6IjE3LjcuMCIsInRva2VuIjoiY2UwZmFjMTk1MDhmNmM4ZjIwMDY2ZDM0NWQzNjBmZDAiLCJUYXJnZXQgT1MgVHlwZSI6IndhdGNob3MiLCJMYW5ndWFnZSI6InN3aWZ0IiwiU3dpZnQgVmVyc2lvbiI6IjQuMiIsIlJlYWxtIFZlcnNpb24iOiIzLjEwLjAifX0=&ip=1, NSLocalizedDescription=can’t load from network, _NSURLErrorRelatedURLSessionTaskErrorKey=(
    "LocalDataTask <4E78EE87-9751-4A16-A8E2-876795969756>.<2>"
), NSUnderlyingError=0x7a01b960 {Error Domain=kCFErrorDomainCFNetwork Code=-2000 "(null)"}, _NSURLErrorFailingURLSessionTaskErrorKey=LocalDataTask <4E78EE87-9751-4A16-A8E2-876795969756>.<2>} [-2000]
```

## Steps to Reproduce
Just try to run a Watch Extension at XCode 10 using Apple Watch series 4 simulator

## Version of Realm and Tooling
Realm framework version: 3.10.0

Xcode version: XCode 10.0

iOS/OSX version: Simulato running on macOS High Sierra 10.13.6

Dependency manager + version: Carthage 0.29

**Comment by tgoyne** [comment]
The network request fails, but the app doesn't crash for me.

**Comment by ggirotto** [comment]
@tgoyne I found the issue, sorry about the inconvenience.

For those who may face a similar issue, in my situation is a VERY specific situation. We have a protocol for our local persistence, which exposes the Realm basic functions (add, object, objects, delete, deleteall, write) and I make Realm conform to this protocol making `extension Realm: LocalPersistenceProtocol { }`. I've found that the functions signatures from the protocol were not equal with Realm signatures, this way when I called the Realm base functions they were not being found (EXC_BAD_ACCESS error was thrown).
